client/src/components/Signup.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+
+import { Button, Card, Form, Input } from 'semantic-ui-react';
+
+import validateUserInput from '../utils/validations/signup';
+import '../styles/form.css';
+
+/**
+ * @class SignUpForm
+ * @extends {React.Component}
+ */
+class SignUpForm extends Component {
+  /**
+   * @description constructor function
+   * @param {any} props
+   * @memberof SignUpForm
+   * @return {void}
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+      usernameError: false,
+      emailError: false,
+      passwordError: false,
+      confirmPasswordError: false,
+      usernameErrorMsg: '',
+      emailErrorMsg: '',
+      passwordErrorMsg: '',
+      confirmPasswordErrorMsg: '',
+      serverError: ''
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  /**
+   *
+   * @param {any} e
+   * @memberof SignUpForm
+   * @returns {void}
+   */
+  onChange(e) {
+    this.setState({
+      [e.target.id]: e.target.value,
+      [`${e.target.id}Error`]: false
+    });
+  }
+
+  /**
+   *
+   * @param {any} e
+   * @memberof SignUpForm
+   * @returns {void}
+  */
+  onSubmit(e) {
+    e.preventDefault();
+
+    const { error, errors } = validateUserInput(this.state);
+
+    if (error) {
+      this.setState({ errors });
+      return;
+    }
+
+    const headers = new Headers({
+      'Content-Type': 'application/json'
+    });
+
+    fetch(`${API_URL}/api/v1/user`, {
+      headers,
+      method: 'POST',
+      body: JSON.stringify({
+        username: this.state.username,
+        email: this.state.email,
+        password: this.state.password
+      })
+    })
+      .then(result => result.json())
```

Comment:
Why do we have two callbacks here,

Reply:
I had to do that since the actual data expected was not available in the first `.then`. It returned a `promise`.

Reply:
Ok...got it!